pyparsing.py

```diff
@@ -104,7 +104,26 @@
 import copy
 import sys
 import warnings
-import re
+regex_found = False
+if sys.version_info >= (2, 0) and sys.version_info < (3,0):
+    import imp
+    try:
+        imp.find_module('eggs')
+        regex_found = True
+    except ImportError:
+        regex_found = False
+if sys.version_info >= (3, 0) and sys.version_info <= (3,3):
+    import importlib
+    regex_loader = importlib.find_loader('regex')
+    regex_found = regex_loader is not None
+if sys.version_info >= (3, 4):
+    from importlib.util import find_spec
+    regex_spec = find_spec("regex")
+    regex_found = regex_spec is not None
+if regex_found:
+    import regex as re
+else:
+    import re
```

Comment:
Can you reduce these lines to:

```python
try:
    import regex as re
except:
    import re
```

Reply:
Not fond of bare `except:`s.  We could do `except ImportError:`, or just `except Exception:` to guard against other exceptions that might crop up in the regex module itself.